### PR TITLE
Normalize package reference for WinUI between C# and C++ templates

### LINKED ISSUE #8486: Logic to choose between WinUI2 and WinUI3 should be in targets, not C# app template
Starting in 0.65 we added the following logic into the C# app template:

```xml
<Choose>
    <When Condition="'$(UseWinUI3)'!='true'">
      <ItemGroup>
        <PackageReference Include="Microsoft.UI.Xaml" >
          <Version>$(WinUI2xVersion)</Version>
        </PackageReference>
      </ItemGroup>
    </When>
    <Otherwise>
      <ItemGroup>
        <PackageReference Include="Microsoft.WinUI">
          <Version>$(WinUI3Version)</Version>
        </PackageReference>
      </ItemGroup>
    </Otherwise>
  </Choose>
```

This logic should really have been in the target files and not added to the template. It'll now be a breaking change to remove but I think it'll be necessary to support the work for PackageReference and community modules building off the nugets.

### PR DESCRIPTION
**Pull request by jonthysell**
## Description

The targets for RNW UWP apps followed different patterns when trying to include WinUI. C++ included a one-liner with the WinUI package name and version props, directly in the app project file. C# ignored the props and used a large MSBuild conditional block directly in the app project file.

This change moves to using the one-liner for both, and puts it into the external targets file. Users who want to specify a different version can override the props (what we want) rather than modifying their project file. I've also moved around some of the other references so that the CsharpApp and CppApp targets are more normalized.

Closes #8486

### Type of Change

- Bug fix (non-breaking change which fixes an issue)

### Why

Having completely different logic for importing the same references into projects is extra overhead to test and maintain. Exposing things in the app template files that we don't want users to change directly increases the odds users break themselves.

Resolves #8486

###### Microsoft Reviewers: [Open in CodeFlow](https://portal.fabricbot.ms/api/codeflow?pullrequest=https://github.com/microsoft/react-native-windows/pull/9372)

### CODE REVIEW

**Review by dannyvv — APPROVED** [review]

File: packages/playground/windows/ExperimentalFeatures.props [hunk 1]
```diff
@@ -8,6 +8,11 @@
   </PropertyGroup>
 
   <PropertyGroup Label="Unpackaged playground-win32" Condition="'$(SolutionName)'=='playground-win32'">
+    <!--
+      WARNING - DO NOT MAKE THE VERSION OF MICROSOFT.UI.XAML BE ANYTHING BUT PRERELEASE.
+      Playground-Win32 is an unpackaged win32 app, and needs to use Microsoft.UI.Xaml from a prerelease,
+      to be able to carry WinUI in-app instead of using the Framework Package
+    -->
     <WinUI2xVersion>2.7.0-prerelease.210913003</WinUI2xVersion>
```
Comment (JunielKatarn) [inline packages/playground/windows/ExperimentalFeatures.props:16]: Is this property needed anymore?

Maybe we can just use `WinUIPackageVersion`?

Comment (jonthysell) [reply packages/playground/windows/ExperimentalFeatures.props:16]: We could, at the risk of `WinUIPackageName` being wrong. Plus I don't think we expect either `WinUIPackageName` or`WinUIPackageVersion` to be set by users, as I don't think they are setup in the targets to only set if blank.

**Comment by jonthysell** [comment]
This PR is still a draft because I literally can't confirm if it's possible to build SampleApps (#9424) or playground (#9436) in VS. Since those don't work now, I can't verify if my PR works or breaks them further.

**Comment by JunielKatarn** [comment]
> 

Please try using MSBuild.

```
MSBuild.exe /restore /p:Platform=x86 path/to/playground.sln
```

File: packages/playground/windows/Playground-win32 (Package)/Playground-win32-packaging.proj [hunk 2]
```diff
@@ -63,5 +63,5 @@
   </PropertyGroup>
   <Import Project="..\..\..\..\vnext\PropertySheets\WinUI.props" />
   <Import Project="$(WapProjPath)\Microsoft.DesktopBridge.targets" />
-  <Import Project="..\packages\$(WinUIPackageName).$(WinUI3Version)\build\Microsoft.WinUI.AppX.targets" />
+  <Import Project="..\packages\$(WinUIPackageName).$(WinUI3Version)\build\Microsoft.WinUI.AppX.targets" Condition="Exists('..\packages\$(WinUIPackageName).$(WinUI3Version)\build\Microsoft.WinUI.AppX.targets')" />
```
Comment (JunielKatarn) [inline packages/playground/windows/Playground-win32 (Package)/Playground-win32-packaging.proj:66]: I this still needed?

Comment (jonthysell) [reply packages/playground/windows/Playground-win32 (Package)/Playground-win32-packaging.proj:66]: You're right, leaving this in causes problems.

**Pull request merged**